### PR TITLE
SALTO-5692: Okta: Upgrade `NetworkZone` deploy definitions to new infra

### PR DESCRIPTION
**Pull request by Nurdok**
See title.

---

_Additional context for reviewer_:
None

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/src/definitions/deploy/deploy.ts [hunk 1]
```diff
@@ -244,38 +218,87 @@ const createCustomizations = (): Record<string, InstanceDeployApiDefinitions> =>
...
+          ],
+          modify: [
+            {
+              request: {
+                endpoint: {
+                  path: '/api/v1/zones/{id}',
+                  method: 'put',
+                },
+              },
+            },
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/definitions/deploy/deploy.ts:248]: 2 suggestions - 
1. should we omit `status` from this request ?
2. should we make this call only if other properties besides `status` changed ? (using `transformForCheck`

File: packages/okta-adapter/src/definitions/deploy/utils/simple_status.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+export const toActionNames: ({
+  change,
+}: definitionUtils.deploy.ChangeAndContext) => (ActionName | AdditionalAction)[] = ({ change }) => {
+  if (isAdditionChange(change)) {
+    // Conditions inside 'activate' and 'deactivate' will determine which one to run, based on the service
+    // response to the 'add' action.
+    return ['add', 'deactivate', 'activate']
+  }
+  if (isModificationChange(change)) {
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/definitions/deploy/utils/simple_status.ts:67]: Adding @shoshana-houri comment from - https://github.com/salto-io/salto/pull/6398#discussion_r1715078920
the `isModificationChange` is redundant here cause we validate it in `isActivationChange` and `isDeactivationChange`

File: packages/okta-adapter/src/definitions/deploy/deploy.ts (hunk 1 above)
Comment (Nurdok) [reply packages/okta-adapter/src/definitions/deploy/deploy.ts:248]: Done.

File: packages/okta-adapter/src/definitions/deploy/utils/simple_status.ts (hunk 2 above)
Comment (Nurdok) [reply packages/okta-adapter/src/definitions/deploy/utils/simple_status.ts:67]: Done.

**Pull request merged**